aesara_hmc/metrics.py

```diff
@@ -0,0 +1,52 @@
+from typing import Callable, Tuple
+
+import aesara.tensor as aet
+import numpy as np
+import scipy.linalg
+import scipy.stats
+from aesara.tensor.var import TensorVariable
+
+
+def gaussian_metric(
+    inverse_mass_matrix: TensorVariable,
+) -> Tuple[Callable, Callable, Callable]:
+    shape = np.shape(inverse_mass_matrix)[0]
+
+    if inverse_mass_matrix.ndim == 1:
+        mass_matrix_sqrt = np.sqrt(np.reciprocal(inverse_mass_matrix))
+        dot, matmul = lambda x, y: x * y, lambda x, y: x * y
+    elif inverse_mass_matrix.ndim == 2:
+        tril_inv = scipy.linalg.cholesky(inverse_mass_matrix)
+        identity = np.identity(shape)
+        mass_matrix_sqrt = scipy.linalg.solve_triangular(tril_inv, identity, lower=True)
```

Comment:
If `inverse_mass_matrix` is a `TensorVariable`, then these `np` functions need to be converted to their corresponding `at` functions, no?

Reply:
That's a good point. I forgot that it can and will be computed by another algorithm upstream; I will switch.